src/DataObject/DataObjectDocument.php

```diff
@@ -636,4 +641,17 @@ public function onRemoveFromSearchIndexes(string $event): void
         }
     }
 
+    public function getChildDocuments(SiteTree $page): array
+    {
+        $docs = [];
+
+        foreach ($page->AllChildren() as $record) {
+            $document = DataObjectDocument::create($record);
+            $docs[$document->getIdentifier()] = $document;
+            $docs = array_merge($docs, $this->getChildDocuments($record));
```

Comment:
Should this rather be
```suggestion
            $docs = array_merge($docs, $record->getDependentDocuments());
```
to respect the same settings as above plus whatever it needs? It shouldn't be just about getting any and all children just because they are child records, right?

Reply:
Good point, I think I utilise the parameter type `getChildDocuments(SiteTree $page)` in this case since this is a recursive call which only works for `SiteTree`

Reply:
There is a logic check in line 459 that explains the use of this method and how it ties with `SiteTree::config()->get('enforce_strict_hierarchy')`

Reply:
On a second thought, I realized your idea and that's great.
Have fixed including additional checks for archived pages.